libs/wallet/src/order-hooks/use-order-cancel.tsx

```diff
@@ -0,0 +1,82 @@
+import { useCallback, useEffect, useState } from 'react';
+import { determineId } from '@vegaprotocol/react-helpers';
+
+import { useVegaTransaction } from '../use-vega-transaction';
+import { useVegaWallet } from '../use-vega-wallet';
+import { useSubscription } from '@apollo/client';
+import type {
+  OrderEvent,
+  OrderEventVariables,
+  OrderEvent_busEvents_event_Order,
+} from './__generated__/OrderEvent';
+import { ORDER_EVENT_SUB } from './order-event-query';
+
+export const useOrderCancel = () => {
+  const { keypair } = useVegaWallet();
+  const { send, transaction, reset: resetTransaction } = useVegaTransaction();
+  const [finalizedOrder, setFinalizedOrder] =
+    useState<OrderEvent_busEvents_event_Order | null>(null);
+  const [id, setId] = useState('');
+
+  // Start a subscription looking for the newly created order
+  useSubscription<OrderEvent, OrderEventVariables>(ORDER_EVENT_SUB, {
+    variables: { partyId: keypair?.pub || '' },
+    skip: !id,
+    onSubscriptionData: ({ subscriptionData }) => {
+      if (!subscriptionData.data?.busEvents?.length) {
+        return;
+      }
+      // No types available for the subscription result
+      const matchingOrderEvent = subscriptionData.data.busEvents[0].event;
+
+      if (matchingOrderEvent && matchingOrderEvent.__typename === 'Order') {
```

Comment:
thought (non-blocking): i wonder if it's worth generating type names as enums or keeping a map in VegaTypes to stop all these magic strings

Reply:
i updated this to use BusEventType.Order

Reply:
Can you undo that please. That is the name of the event type which is not nesseacrily the name of the entity in GQL. Leave as is, I just meant something I think we should look into, as this problem is across the whole codebase. Probably to use the introspection query to create an enum of type names or something but not something for now :)

Reply:
ok, alright.